frontend/src/components/object/ObjectPermission.vue

```diff
@@ -25,19 +27,40 @@ const { getMappedObjectToUserPermissions } = storeToRefs(permissionStore);
 const showSearchUsers: Ref<boolean> = ref(false);
 
 // Actions
+const removeManageAlert = useAlert('Warning', 'Cannot remove last user with MANAGE permission.');
+
 const cancelSearchUsers = () => {
   showSearchUsers.value = false;
 };
 
 const removeObjectUser = (userId: string) => {
-  permissionStore.removeObjectUser(props.objectId, userId);
+  const managers = getMappedObjectToUserPermissions.value.filter( (x: UserPermissions) => x.manage );
+  if( managers.length === 1 && managers[0].userId === userId ) {
+    removeManageAlert.show();
+  }
+  else {
+    permissionStore.removeObjectUser(props.objectId, userId);
+  }
 };
 
 const updateObjectPermission = (value: boolean, userId: string, permCode: string) => {
   if (value) {
     permissionStore.addObjectPermission(props.objectId, userId, permCode);
   } else {
-    permissionStore.deleteObjectPermission(props.objectId, userId, permCode);
+    const managers = getMappedObjectToUserPermissions.value.filter( (x: UserPermissions) => x.manage );
+
+    // Disallow removable of final MANAGE permission
+    if( permCode === Permissions.MANAGE && !managers.length ) {
+      removeManageAlert.show();
+
+      // Set the value back as clicking will automatically change it
+      const perm: UserPermissions = getMappedObjectToUserPermissions.value
+        .find( (x: UserPermissions) => x.userId === userId ) as UserPermissions;
+      perm.manage = true;
+    }
+    else {
+      permissionStore.deleteObjectPermission(props.objectId, userId, permCode);
+    }
```

Comment:
I can't quite tell from just visual inspection, but does this logic also factor in situations where a user may have manage permission on a bucket level, but for the object they're looking at, there's no explicit permissions bound to it? May need to think further on edge cases for this, but it may be permssible to allow an object to have no users bound to it at all, as users will have a way to escape hatch it and reapply perms if needed by virtue of having bucket level permissions cascading downwards?

Reply:
I believe we should allow someone with MANAGE perm (either at obj or bucket level) to remove whoever they want as long as at least one person is always left who has the MANAGE perm.

To get to object permissions modal, the current user must have MANAGE perm at either bucket-level or object-level.
I think we only need to prevent the current user from de-selecting MANAGE (or removing user) for **themself** and they are the last user in the object permissions list with object-level MANAGE perm.